### PR TITLE
[DOCS] Edits ML hyperparameter descriptions

### PR DESCRIPTION
**Pull request by lcawl**
Related to https://github.com/elastic/elasticsearch/pull/67950 and https://github.com/elastic/kibana/issues/90738

This PR edits the descriptions of the machine learning data frame analytics hyperparameters, which are used in several APIs.

### CODE REVIEW

File: docs/reference/ml/ml-shared.asciidoc [hunk 1]
```diff
@@ -537,24 +537,26 @@ the detectors in the `analysis_config`, starting at zero.
...
-the most. This parameter multiplies a term based on tree depth in
-the regularized loss. Higher values result in shallower trees
-and faster training times. Values should be greater than or equal
-to zero. By default, this value is calculated during hyperparameter optimization.
+Advanced configuration option. {ml-cap} uses loss guided tree growing, which
+means that the decision trees grow where the regularized loss decreases most
+quickly. This parameter affects loss calculations by acting as a multiplier of 
+the tree depth. Higher alpha values result in shallower trees and faster 
+training times.
+//TBD: but potentially reduce accuracy?
```
Comment (dimitris-athanasiou) [inline docs/reference/ml/ml-shared.asciidoc:545]: Unfortunately this is quite nuanced and very much data-dependent. I think we might be better off without mentioning anything more here for now. @valeriy42 is working on an example notebook that goes into more detail on explaining hyperparameters. When that is ready we can probably link to that for users that want to delve into the more nuanced details.

**Review by dimitris-athanasiou — COMMENTED** [review]
LGTM Much nicer wording!

**Review by szabosteve — APPROVED** [review]
Thanks for these changes. LGTM!

File: docs/reference/ml/ml-shared.asciidoc (hunk 1 above)
Comment (lcawl) [reply docs/reference/ml/ml-shared.asciidoc:545]: Sounds good to me, thanks!  I'll remove that TBD

**Pull request merged**